File: 1480-running-sum-of-1d-array/1480-running-sum-of-1d-array.py
class Solution(object):
    def runningSum(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        sum = [nums[0]]
        for i in range(1,len(nums)):
            sum.append(sum[i-1]+nums[i])
            
        return sum

#         sum = 0
#         output = []
#         for i in range(len(nums)):
#             sum = sum+nums[i]
#             output.append(sum)
        
#         return output
    

File: 1480-running-sum-of-1d-array/1480-running-sum-of-1d-array.go
func runningSum(nums []int) []int {
    
//     sum := []int{nums[0]}
//     length := len(nums)
    
//     for i := 1; i < length; i++ {
//         sum = append(sum, sum[i-1]+nums[i])
//     }
    
//     return sum
    
// 2nd ...
    
//     sum := make([]int, len(nums))
    
//     sum[0] = nums[0]
    
//     for i := 1; i < len(nums); i++ {
//         sum[i] = sum[i-1]+nums[i]
//     }
    
//     return sum

// 3rd ...
    
    for i := 1; i < len(nums); i++ {
        nums[i] += nums[i-1]
    }
    
    return nums
    
}